### PR TITLE
Magic fix for "connection was not ready".

### PR DESCRIPTION
**Pull request by meowjesty**
- Issue: #1302

1. Applies a magic fix that should help with hyper's "connection was not ready" error.
2. Adds some more logs for HTTP stealer in layer.

### CODE REVIEW

File: mirrord/layer/src/tcp_steal/http.rs [hunk 1]
```diff
@@ -84,6 +85,7 @@ pub(super) trait HttpV: Sized {
 
     /// Sends the [`HttpRequest`] through `Self::sender`, converting the response into a
     /// [`HttpResponse`].
+    #[tracing::instrument(level = "trace", skip(self))]
```
Comment (aviramha) [inline mirrord/layer/src/tcp_steal/http.rs:88]: skip `HttpRequest` or it's without data?

File: mirrord/layer/src/tcp_steal/http.rs [hunk 2]
```diff
@@ -101,6 +103,8 @@ pub(super) trait HttpV: Sized {
 
         // Retry once if the connection was closed.
         if let Err(HttpForwarderError::ConnectionClosedTooSoon(request)) = response {
+            trace!("Request {request:#?} connection was closed too soon, retrying once!");
```
Comment (aviramha) [inline mirrord/layer/src/tcp_steal/http.rs:106]: same as above?

**Review by aviramha — APPROVED** [review]
LGTM besides few nitpicks!

File: mirrord/layer/src/tcp_steal/http.rs (hunk 1 above)
Comment (meowjesty) [reply mirrord/layer/src/tcp_steal/http.rs:88]: Yeah, the `body` is logged as just `bytes (length): 100` for example.

**Pull request merged**